api/v1/services/UserService.js

```diff
@@ -118,3 +118,13 @@ module.exports.resetPassword = function (user, password) {
 			return updated.save();
 		});
 };
+
+module.exports.deleteUser = function (user) {
+	if(user.hasRoles(utils.roles.ALL, false)) {
+		var message = "The provided user has a role and can not be deleted";
+		var source = "UserRole";
```

Comment:
This isn't a thing that the client knows about right? We should not supply a source.